__test__/authorizer.test.js

```diff
@@ -0,0 +1,106 @@
+const {
+  authorizeStagingToken,
+  authorizeProductionToken
+} = require("authorizer");
+const nock = require("nock");
+
+describe("Verification", () => {
+  beforeEach(() => {
+    nock("https://auth-next.quicktravel.com.au")
+      .get("/.well-known/jwks.json")
+      .reply(200, {
+        keys: [
+          {
+            kty: "RSA",
+            n:
+              "517vaFTRKc98hCKiyF_u4GCiZuRKA4xX4x4lWSXB38w0FasyqwHhl686L6AzGZQPolEPyY-fF0xn3huricBKPgteYh35ij3hUwMZBDaZFCESlE4H0s8u9eONp2Kfb2ER3ILaRMM-RsUImnGUiNUy9ZW4k3zI4_wIn3YC0UX5WF83QhvnxSFyE8IQdnvsp6mn7Bv0lQk43pfQlvBudEjXEGyKBzEDUzinkp_gvfIxKdZ_Kfnu70MWndSl0sa4x7csgfV5rh4O3VnHIk7PHARdCWiOQvj8ze_pDKpM6Vy28MArk2OCb1vHT28ol2qxbG8mlPv_bZdsBGrT1-65pXn-Sw",
+            e: "AQAB",
+            kid:
+              "00fa1d3f44830d217ca99e6cd021981a38c2b6e18325412c9e378605ca976fc6"
+          }
+        ]
+      });
+
+    nock("https://auth.quicktravel.com.au")
+      .get("/.well-known/jwks.json")
+      .reply(200, {
+        keys: [
+          {
+            kty: "RSA",
+            n:
+              "517vaFTRKc98hCKiyF_u4GCiZuRKA4xX4x4lWSXB38w0FasyqwHhl686L6AzGZQPolEPyY-fF0xn3huricBKPgteYh35ij3hUwMZBDaZFCESlE4H0s8u9eONp2Kfb2ER3ILaRMM-RsUImnGUiNUy9ZW4k3zI4_wIn3YC0UX5WF83QhvnxSFyE8IQdnvsp6mn7Bv0lQk43pfQlvBudEjXEGyKBzEDUzinkp_gvfIxKdZ_Kfnu70MWndSl0sa4x7csgfV5rh4O3VnHIk7PHARdCWiOQvj8ze_pDKpM6Vy28MArk2OCb1vHT28ol2qxbG8mlPv_bZdsBGrT1-65pXn-Sw",
+            e: "AQAB",
+            kid:
+              "00fa1d3f44830d217ca99e6cd021981a38c2b6e18325412c9e378605ca976fc6"
+          }
+        ]
+      });
+  });
+
+  test("Throws exception when no token is passed", async () => {
+    const event = {};
+    await expect(authorizeStagingToken(event, {})).rejects.toThrow(
+      "Unauthorized"
+    );
+  });
+
+  test("Throws exception when no bearer token is passed", async () => {
+    const event = {
+      authorizationToken: "test-auth"
+    };
+
+    await expect(authorizeStagingToken(event, {})).rejects.toThrow(
+      "Unauthorized"
+    );
+  });
+
+  test("Throws exception when issueing server can not be reached", async () => {
+    nock("https://auth-next.quicktravel.com.au")
+      .get("/.well-known/jwks.json")
+      .reply(500, {});
+    const event = {};
+
+    await expect(authorizeStagingToken(event, {})).rejects.toThrow(
+      "Unauthorized"
+    );
+  });
+
+  test("Throws exception when token is invalid", async () => {
+    const event = {
+      authorizationToken: "Bearer test-auth-invalid"
+    };
+
+    await expect(authorizeStagingToken(event, {})).rejects.toThrow(
+      "Unauthorized"
+    );
+  });
+
+  test("Validates correct token", async () => {
```

Comment:
I wouldn't mind seeing one more test for an out of date token

Reply:
Added a test for expired tokens as well